resources/js/components/SignIn.js

```diff
@@ -1,292 +1,250 @@
-import DialogTitle from "@mui/material/DialogTitle";
-import CircularProgress from "@mui/material/CircularProgress";
-import IconButton from "@mui/material/IconButton";
-import InputAdornment from "@mui/material/InputAdornment";
-import Visibility from "@mui/icons-material/Visibility";
-import VisibilityOff from "@mui/icons-material/VisibilityOff";
+import Avatar from '@mui/material/Avatar';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Link from '@mui/material/Link';
+import Paper from '@mui/material/Paper';
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import CircularProgress from '@mui/material/CircularProgress';
+import IconButton from '@mui/material/IconButton';
+import InputAdornment from '@mui/material/InputAdornment';
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
 
-import Footer from "./layouts/Footer";
+import Footer from './layouts/Footer';
 import {
-    freshState,
-    signIn,
-    validateEmail,
-    validatePassword,
-    showPassword,
-    disableSubmit,
-} from "../actions/userAuthActions";
+  freshState,
+  signIn,
+  validateEmail,
+  validatePassword,
+  showPassword,
+  disableSubmit
+} from '../actions/userAuthActions';
 
 const SignIn = (props) => {
-    const navigate = useNavigate();
+  const navigate = useNavigate();
 
-    // When navigated to this component, re-initialize the component state
-    useEffect(() => {
-        props.freshState();
-        /* 
-            When navigated to this component, the password should be cleared
-            but when you put password in this component and try to navigate to
-            Sign Up, the password field has values. This checks if password has
-            values on first render then clear the values.
-        */
-    }, []);
+  // Check requestError, if false then navigate to dashboard page
+  useEffect(() => {
+    if (props.requestError === false) {
+      const timer = setTimeout(() => {
+        navigate('/dashboard');
+      }, 2500);
+      return () => clearTimeout(timer);
+    }
+  }, [props.requestError]);
 
-    // Check requestError, if false then navigate to dashboard page
-    useEffect(() => {
-        if (props.requestError === false) {
-            const timer = setTimeout(() => {
-                navigate("/dashboard");
-            }, 2500);
-            return () => clearTimeout(timer);
-        }
-    }, [props.requestError]);
+  // This will run, if there is changes in the state
+  useEffect(() => {
+    // Enable the submit button if no errors in fields
+    if (props.isValidEmail === true && props.isValidPassword === true) {
```

Comment:
here as well, no need to explicitly check if `true` or `false` if variable's value is boolean.

```suggestion
    if (props.isValidEmail && props.isValidPassword) {
```

Reply:
Removed the `=== true` and it is still functional. This is resolved